storage/innobase/xtrabackup/src/xbcloud/xbcloud.cc

```diff
@@ -1038,10 +1038,16 @@ bool xbcloud_put(Object_store *store, const std::string &container,
 @return true in case of success or false otherwise */
 bool chunk_name_to_file_name(const std::string &chunk_name,
                              std::string &file_name, my_off_t &idx) {
-  if (chunk_name.size() < 22 && chunk_name[chunk_name.size() - 21] != '.') {
+  if (chunk_name.size() < 22 || chunk_name[chunk_name.size() - 21] != '.') {
```

Comment:
This was noted in another #1538. Technically, it is a problem but not seen in the real world because of the minimum prefix length uploaded.

We can accept this.